plugin/src/main/java/org/opensearch/ml/model/MLModelCacheHelper.java

```diff
@@ -537,6 +537,21 @@ public String[] getWorkerNodes(String modelId) {
         return modelCache.getWorkerNodes();
     }
 
+    /**
+     * Get target worker nodes of model.
+     *
+     * @param modelId model id
+     * @return array of node id; return null if model not exists in cache
+     */
+    public String[] getTargetWorkerNodes(String modelId) {
+        MLModelCache modelCache = modelCaches.get(modelId);
+        if (modelCache == null) {
+            return null;
+        }
+        return modelCache.getTargetWorkerNodes();
```

Comment:
We should consider deploy to all nodes case. 
If deploy to all nodes is try and target work nodes may be [node1, node2] on day1. 
Then day2, user add one more nodes to cluster, now the target work nodes should be [node1, node2, node3], so we can deploy to all nodes.
I think the work node from cache will still be [node1, node2], right ?

Reply:
From what I see the synup job only syncs up the worker nodes, but not the target worker nodes. This is another enhancement needed. Both the target worker nodes and running worker nodes needs to be up-to-date in the memory so it can cover all kinds of cases.